libr/core/canal.c

```diff
@@ -4275,25 +4275,29 @@ R_API RCoreAnalStats* r_core_anal_get_stats(RCore *core, ut64 from, ut64 to, ut6
 		piece = (S->vaddr - from) / step;
 		as->block[piece].symbols++;
 	}
-	RList *metas = r_meta_enumerate (core->anal, -1);
-	RAnalMetaItem *M;
-	r_list_foreach (metas, iter, M) {
-		if (M->from < from || M->to > to) {
-			continue;
-		}
-		piece = (M->from - from) / step;
-		switch (M->type) {
-		case R_META_TYPE_STRING:
-			as->block[piece].strings++;
-			break;
-		case R_META_TYPE_COMMENT:
-			as->block[piece].comments++;
-			break;
+	RPVector *metas = to >= from ? r_meta_get_all_intersect (core->anal, from, to - from, R_META_TYPE_ANY) : NULL;
```

Comment:
You can replace the ternary operator with an if and an early return, the rest of the function wont do a thing if the check fails

Reply:
Normally I would agree about the early return, but here it is only one part of the function that might fail, not the entire function. Someone might add another type of info after the meta stuff that shouldn't be skipped if the meta check fails and early return would break that.
But the `to >= from` condition is actually wrong. `r_meta_get_all_intersect() asserts for size >= 1`.